apiserver/facades/client/application/application.go

```diff
@@ -457,11 +459,12 @@ func (c caasDeployParams) precheck(
 	registry storage.ProviderRegistry,
 	caasBroker CaasBrokerInterface,
 ) error {
-	if len(c.attachStorage) > 0 {
+	if len(c.attachStorage) > 0 && !featureflag.Enabled(feature.K8SAttachStorage) {
```

Comment:
Strictly we could have the feature flag check here (since it would cover the pylibjuju callers).
But adding it to the CLI instead ensures the check is done before an api call is made and a better user facing error message is surfaced. The flag will be short lived anyway.